File: Leetcode/JS/Leetcode400-第N位数字.js
var findNthDigit = function(n) {
    let cnt = 1,base = 9,s = 1;
    while(n > cnt * base){
        n -= cnt * base;
        cnt++;
        base *= 10;
        s *= 10;
    }
    s += Math.floor((n + cnt - 1) / cnt) - 1;
    n = n % cnt ? n % cnt : cnt;
    return String(s).codePointAt(n - 1) - '0'.codePointAt(0);
};

File: Leetcode/sbgo/Leetcode400-第N位数字.go
package sbgo

import "strconv"

func findNthDigit(n int) int {
	cnt, base, s := 1, 9, 1
	for n > cnt*base {
		n -= base * cnt
		cnt++
		base *= 10
		s *= 10
	}
	s += (n+cnt-1)/cnt - 1
	if n%cnt == 0 {
		n = cnt
	} else {
		n = n % cnt
	}
	return int(strconv.Itoa(s)[n-1] - '0')
}
